datastore/threads/databuffer.nim

```diff
@@ -50,15 +51,17 @@ proc new*[T: byte | char](tp: type DataBuffer, data: openArray[T]): DataBuffer =
   ##
   result = DataBuffer.new(data.len)
   if data.len() > 0:
-    copyMem(result[].buf, unsafeAddr data[0], data.len)
+    # TODO: we might want to copy data, otherwise the GC might
+    # release it on stack-unwind
+    copyMem(result[].buf, baseAddr data, data.len)
 
 converter toSeq*(self: DataBuffer): seq[byte] =
   ## convert buffer to a seq type using copy and either a byte or char
   ##
 
   result = newSeq[byte](self.len)
   if self.len() > 0:
-    copyMem(addr result[0], unsafeAddr self[].buf[0], self.len)
+    copyMem(addr result[0], addr self[].buf[0], self.len)
```

Comment:
Ah good! I was thinking about that this weekend.

Reply:
yeah, but maybe this should also be switched to `baseAddr`? Not sure.

Reply:
Or the other way around, switch `baseAddr` to `addr`?

Reply:
Imo, `addr` is probably safer, `baseAddr` does a straight up `cast`.